woocommerce/woocommerce-ios
# [Mobile Payments] Refactor cpp eligibility logic into Yosemite

Sorry for the larger PR, hopefully you can still properly review it, otherwise I can try to split it in smaller PRs :) 

Closes: #7004


### Description

Before we had the order card payment present eligibility logic in the WooCommerce target. With this PR we move that to Yosemite, where business logic feels more natural to stay. Likewise, following other examples like the checking for the shipping labels eligibility, we create an store (`OrderCardPresentPaymentEligibilityStore`) to implement the logic, requiring just an `orderID`, `siteID`, and configuration as in other store cases. In that way, we remove the products parameter that was necessary before to check whether the order products contain any subscription because now we retrieve those from storage in the store action handling method.
After moving this to `Yosemite`, we adapt Order Details and Simple Payment Methods classes to this new implementation, removing the results controller from `SimplePaymentsMethodsViewModel`. Furthermore, we take also take into consideration that when a payment is collected we should refresh the order details UI so the button is not shown anymore.
Also, I remove some methods from `OrderPaymentDetailsViewModel` that were not used anymore, as well as moved some to `Yosemite` so we can reuse them.

### Notes

I removed the unit tests relative to  `checkCardPresentPaymentEligibility` in https://github.com/woocommerce/woocommerce-ios/pull/7063/commits/902cfd1454539cdd535e664803ae8b4f01d53e06 because we couldn't inject/mock the configuration loader and failed.. To keep that PR small, I create this [issue](https://github.com/woocommerce/woocommerce-ios/issues/7064) to accomplish that.

### Testing instructions

No UI changes whatsoever are added here, so the collect payment button is shown in the same circumstances as before:
- Order is not paid totally
- The order total and the remaining amount to collect is the same
- Currency is eligible for card payment
- Order payment method is eligible for card present payment `.booking, .cod, .woocommercePayments, .stripe, .none`
- Country is supported for card present payment

More information on this table (from @joshheald  in https://github.com/woocommerce/woocommerce-ios/pull/7046):

Order status | Payment status | Expected button state on CPP store | Expected button state on non-CPP store
---|---|---|---
Complete | Paid | Issue Refund | Issue Refund
On hold | Paid | Issue Refund | Issue Refund
Processing | Paid | Issue Refund | Issue Refund
On hold | Unpaid | Collect Payment | No button*
Processing | Unpaid | Collect Payment | No button*
Pending payment | Unpaid | Collect Payment | No button*
Complete | Unpaid | No button | No button
Refunded | Paid | No button | No button
Complete | Paid and part refunded | Issue Refund | Issue Refund
Complete | Paid and fully refunded | No button | No button

We also have to check that:
- The button disappears when a payment is collected successfully
- Taking into account the history here, we have to make sure that no UI flickering is added to the Order Details Screen

### Screenshots



---
